cluster-agent/utils/argocd_create.go

```diff
@@ -0,0 +1,453 @@
+				NonResourceURLs: []string{},
+			},
+		},
+	}
+
+	// Delete the resources, if they exists
+
+	_, errOnGet = kubeClientSet.RbacV1().ClusterRoles().Get(context.Background(), clusterRole.Name, metav1.GetOptions{})
+	if errOnGet == nil {
+		errOnDelete := kubeClientSet.RbacV1().ClusterRoles().Delete(context.Background(), clusterRole.Name, metav1.DeleteOptions{PropagationPolicy: &policy})
+		if errOnDelete != nil {
+			return fmt.Errorf("Error on DELETE %v", errOnDelete)
+		}
+
+	}
+	if err := k8sClient.Create(context.Background(), &clusterRole); err != nil {
+		if !apierr.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create clusterRole %q in namespace %q: %v", clusterRole.Name, clusterRole.Namespace, err)
+		}
+		return fmt.Errorf("clusterRole %q already exists in namespace %q", clusterRole.Name, clusterRole.Namespace)
+	}
+	log.Printf("clusterRole %q created in namespace %q", clusterRole.Name, clusterRole.Namespace)
+
+	clusterRoleBinding := rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "argocd-manager-role-binding",
+			Namespace: "kube-system",
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "argocd-manager-role",
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "argocd-manager",
+				Namespace: "kube-system",
+			},
+		},
+	}
+
+	// debug : cluster role binding doesn't respond to the below code, need to delete manually using kubectl.
+	_, errOnGet = kubeClientSet.RbacV1().ClusterRoleBindings().Get(context.Background(), clusterRoleBinding.Name, metav1.GetOptions{})
+	if errOnGet == nil {
+		errOnDelete := kubeClientSet.RbacV1().ClusterRoleBindings().Delete(context.Background(), clusterRoleBinding.Name, metav1.DeleteOptions{PropagationPolicy: &policy})
+		if errOnDelete != nil {
+			return fmt.Errorf("Error on DELETE %v", errOnDelete)
+		}
+	}
+	if err := k8sClient.Create(context.Background(), &clusterRoleBinding); err != nil {
+		if !apierr.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create clusterRoleBinding %q in namespace %q: %v", clusterRoleBinding.Name, clusterRoleBinding.Namespace, err)
+		}
+		return fmt.Errorf("clusterRoleBinding %q already exists in namespace %q", clusterRoleBinding.Name, clusterRoleBinding.Namespace)
+	}
+	log.Printf("clusterRoleBinding %q created in namespace %q", clusterRoleBinding.Name, clusterRoleBinding.Namespace)
+
+	// 	apiVersion: v1
+	// kind: Secret
+	// metadata:
+	//   name: my-cluster-secret
+	//   labels:
+	//     argocd.argoproj.io/secret-type: cluster
+	// type: Opaque
+	// stringData:
+	//   name: my-cluster
+	//   server: $API_URL
+	//   config: |
+	//     {
+	//       "bearerToken": "$SECRET_TOKEN",
+	//       "tlsClientConfig": {
+	//         "insecure": true
+	//       }
+	//     }
+
+	secretResource, secretErr := kubeClientSet.CoreV1().Secrets(secret.Namespace).Get(context.Background(), secret.Name, metav1.GetOptions{})
```

Comment:
I believe we need to use a `wait.Poll(...)` here, similar to what the [gist](https://gist.github.com/jgwest/64cdc63a978324958ab8ac91aea74700) was doing on line 86 to 96. In the case of the gist, it was using a for loop, but in Go a `wait.Poll` would be the preferred idiom.

It would be, something like this, I believe:
```golang
wait.Poll(intervals, func() bool {
    // get the secret
    // check if the secret has a token field, which is non-empty
    // return true (indicating success!) if the secret exists and has the token, otherwise false
})
```
That way, the code will keep waiting for the secret to exist. (otherwise, it's a race condition)